soc/arm/ambiq/apollo4x/soc.c

```diff
@@ -16,7 +17,15 @@
 
 LOG_MODULE_REGISTER(soc, CONFIG_SOC_LOG_LEVEL);
 
+#define AMBIQ_UART_PWRCTRL_ID(uart) AM_HAL_PWRCTRL_PERIPH_UART ## uart
+#define AMBIQ_UART_PWRCTRL_ENABLE(idx) \
+	am_hal_pwrctrl_periph_enable(AMBIQ_UART_PWRCTRL_ID(idx));
```

Comment:
DT instances order is not guaranteed, ie, inst 0 may not be uart0

Reply:
Added a dt property in the Ambiq-specific UART binding that denotes the instance number: https://github.com/zephyrproject-rtos/zephyr/blob/4aace128e0d88e660dcdc4a4473f8736927196d0/dts/bindings/serial/ambiq%2Cuart.yaml#L17-L20